README.md

```diff
@@ -47,81 +73,376 @@ The following sections specify the required and optional content for `eaw.modinf
       "Singleplayer"
     ]
   },
-  "custom":{
-    "my_custom_key": "My important additional value."
-  }
+  "custom": [
+    {
+      "my_custom_key": "My important additional value."
+    }
+  ]
 }
 ```
 
-### The `"name"` Tag [REQUIRED]
+## The `"modinfo"` Type
+
+### The `"name"` Property
+
+**Level:** **REQUIRED**
+
+**Data Type**: `String`  
+
+**Data Semantics**: Unique Text 
+
+**Description:**
+
+This property specifies the fully qualified mod name, e.g. "Republic at War", "Thrawn's Revenge: Imperial Civil War", "Awakening of the Rebellion" or "Yuuzhan Vong at War".
+
+This property **must** also be present for variant modinfos that aim for overriding the main `modinfo.json`
+
+### The `"summary"` Property
 
-This tag specifies the fully qualified mod name, e.g. "Republic at War", "Thrawn's Revenge: Imperial Civil War", "Awakening of the Rebellion" or "Yuuzhan Vong at War"
+**Level:** *OPTIONAL*
 
-### The `"summary"` Tag [OPTIONAL]
+**Data Type**: `String`  
 
-This tag allows you to include a short summary about the mod supporting Steam-flavoured BBCode.
+**Data Semantics**: Text 
 
-### The `"icon"` Tag [OPTIONAL]
+**Description:**
 
-The relative path to the mod's icon file.
+This property allows you to include a short summary about the mod supporting Steam-flavoured BBCode.
 
-### The `"version"` Tag [OPTIONAL]
+### The `"icon"` Property
+
+**Level:** *OPTIONAL*
+
+**Data Type**: `String`  
+
+**Data Semantics**: File System Path 
+
+**Description:**
+
+The path to the mod's icon file **relative** to the mod's root directory or an **absolute** path.
+
+### The `"version"` Property
+
+**Level:** *OPTIONAL*
+
+**Data Type**: `String`  
+
+**Data Semantics**: 4 Digit Semantic Version 
+
+**Description:**
 
 The mod's version according to the extended semantic versioning: [Semantic Versioning](https://semver.org/) that also supports a fourth digits for build increments, etc. and suffixes (e.g. `"-rc1"`).
 
-### The `"dependencies"` Tag [OPTIONAL]
+### The `"dependencies"` Property
+
+**Level:** *OPTIONAL*
+
+**Data Type**: [`modreference`](#the-modreference-type)`[]`
+
+**Data Semantics**: Ordered List of Objects 
+
+**Description:**
+
+The `dependencies` container holds an ordered list of [`"modreference type"`](#the-modreference-type)s this mod relies on.
 
-The `dependencies` container holds an ordered list of other mods instances this mod relies on.
-The first item of the list is the most base mod, every `n+1` mod is an sub mod of `n`. The mod of the this modinfo.json file must not be listed here!
 The list is either absent from the `modinfo.json` or contains at least one item.
 
-#### The `"dependencies.modtype"` Tag
+The first mod in the list resembles the closest dependeny. Every `n+1` item is another dependeny of the target mod.
+If a dependency has dependencies itself these **must not** be added to the list.
+
+**The mod of the this modinfo.json file must not be listed here because that would result in a cycle!**
```

Comment:
So the mod described by the file itsef?
```suggestion
**The mod described by the current `modinfo.json` file must not be listed here, including it would result in a cycle!**
```